### PR TITLE
PQC: Add hybrid groups x25519/ML-KEM-768 and secp256r1/ML-KEM-768

### PR DESCRIPTION
**Pull request by reneme**
### Pull Request Dependencies

* #4389 - which contains tests for x25519/ML-KEM
* #4391 - fixing an issue in ML-KEM when running BoGo with sanitizers

### Description

Those groups are described in [draft-kwiatkowski-tls-ecdhe-mlkem](https://datatracker.ietf.org/doc/draft-kwiatkowski-tls-ecdhe-mlkem/02/) and the code points are [registered by IANA](https://www.iana.org/assignments/tls-parameters/tls-parameters.xhtml#tls-parameters-8).

### CODE REVIEW

**Review by randombit — APPROVED** [review]
:rocket:

**Review by FAlbertDev — APPROVED** [review]

**Comment by reneme** [comment]
Thanks for the reviews. I'm holding back on merging that because I'd like to update bogo (which I believe has test cases for those groups already).

**Comment by reneme** [comment]
Enabling relevant BoGo tests revealed something that we might want to have a look at here (or independently):

```
  FAILED (CurveTest-Invalid-MLKEMEncapKeyNotReduced-Server-MLKEM-TLS13)
  bad error (wanted ":BAD_ECPOINT:" / "remote error: illegal parameter"): local error "EOF", child error "signal: aborted (core dumped)", stdout:
  
  stderr:
  False assertion mapped_coeff <= range in unpack @/home/runner/work/botan/botan/build/build/include/internal/botan/internal/pqcrystals_encoding.h:206
```

File: src/bogo_shim/bogo_shim.cpp [hunk 1]
```diff
@@ -998,12 +998,18 @@ class Shim_Policy final : public Botan::TLS::Policy {
...
+               }
+
                // TODO: once `TLS::Policy::key_exchange_groups()` contains it by
                //       default, remove this explicit check.
-               if(group == Botan::TLS::Group_Params::HYBRID_X25519_KYBER_768_R3_OQS) {
+               //
+               // See: https://github.com/randombit/botan/pull/4305
+               if(group == Botan::TLS::Group_Params::HYBRID_X25519_ML_KEM_768) {
                   groups.push_back(group);
                }
```
Comment (reneme) [inline src/bogo_shim/bogo_shim.cpp:1014]: See also: https://github.com/randombit/botan/pull/4305#discussion_r1805118302

**Review by randombit — APPROVED** [review]

**Pull request merged**